akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala

```diff
@@ -55,28 +54,10 @@ sealed trait HttpEntity extends jm.HttpEntity {
    * Collects all possible parts and returns a potentially future Strict entity for easier processing.
    * The Future is failed with an TimeoutException if the stream isn't completed after the given timeout.
    */
-  def toStrict(timeout: FiniteDuration)(implicit fm: Materializer): Future[HttpEntity.Strict] = {
-    def transformer() =
-      new TimerTransformer[ByteString, HttpEntity.Strict] {
-        var bytes = ByteString.newBuilder
-        scheduleOnce("", timeout)
-
-        def onNext(element: ByteString): immutable.Seq[HttpEntity.Strict] = {
-          bytes ++= element
-          Nil
-        }
-
-        override def onTermination(e: Option[Throwable]): immutable.Seq[HttpEntity.Strict] =
-          HttpEntity.Strict(contentType, bytes.result()) :: Nil
-
-        def onTimer(timerKey: Any): immutable.Seq[HttpEntity.Strict] =
-          throw new java.util.concurrent.TimeoutException(
-            s"HttpEntity.toStrict timed out after $timeout while still waiting for outstanding data")
-      }
-
-    // TODO timerTransform is meant to be replaced / rewritten, it's currently private[akka]; See https://github.com/akka/akka/issues/16393
-    dataBytes.via(Flow[ByteString].timerTransform(transformer).named("toStrict")).runWith(Sink.head)
-  }
+  def toStrict(timeout: FiniteDuration)(implicit fm: Materializer): Future[HttpEntity.Strict] =
```

Comment:
I guess what we need here is a simple generic `aggregateBytes(timeout, maxBytes)` Flow so that this can get

```scala
dataBytes.via(aggreateBytes(timeout, maxBytes)).map(HttpEntity.Strict(contentType, _))
```

I guess the generic `aggregateBytes` flow could even be more generic by allow it to work on every collection that supports `++` and `size`, or a builder that support `append`, `size`, and `result`.